### PR TITLE
Force typeMap['root'] to BSONDocument

### LINKED ISSUE #8: TypeMap root=array not supported
I was experimenting with this bundle to store the failed messages in MongoDB, and its was very easy to achieve. Thank you so much!

I did notice the following
 
```
In Connection.php line 179:
                                                                                                                                                  
  [TypeError]                                                                                                                                     
  Return value of Facile\MongoDbMessenger\Transport\Connection::find() must be an 
instance of MongoDB\Model\BSONDocument or null, array returned
```

For historical reasons, our typeMap is:
``` 
[
        'root' => 'array',
        'document' => 'array',
        'array' => 'array',
    ];
```

I got around this issue by changing the Connection.php file to 
```
    public function find(string $id): ?BSONDocument
    {
        $typeMap = $this->collection->getTypeMap();
        $typeMap['root'] = BSONDocument::class;

        return $this->collection->findOne(['_id' => new ObjectId($id)], ['typeMap' => $typeMap]);
    }

// Similar change is needed in get() and findBy()
```

but not sure if that is a good solution? What do you think?

### PR DESCRIPTION
**Pull request by starred-gijs**
To prevent BC breaks, force the root to BSONDocument...
See https://github.com/facile-it/mongodb-bundle/issues/116

Not sure if this is a valid solution??

### CODE REVIEW

**Comment by Jean85** [comment]
FTR I moved the original issue to https://github.com/facile-it/mongodb-bundle/issues/116, because it could be a low-level issue in the bundle, not simply in this transport.

Maybe we could apply this same solution there..

**Comment by starred-gijs** [comment]
I **do** use that bundle with the typeMap config set to array. In that bundle it's no issue because there is no return typehint set to `BSONDocument` [search](https://github.com/facile-it/mongodb-bundle/search?q=BSONDocument). 
This messenger-transport `Connection` class does, so therefor I changed the specific read options only for these method calls.

ps: tests broken, but I can fix them if this solution is agreed upon

**Comment by Jean85** [comment]
Oh you're totally right, my mistake!

Yes, the solution is fine by me then; please add a changelog entry too.

**Comment by starred-gijs** [comment]
FYI: just wanted to mention that I have been running this version on production without any issues.

Any more actions needed from my side?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## 1.3.0 (TBA)
+* Force typeMap['root] to BSONDocument ([#8](https://github.com/facile-it/mongodb-messenger-transport/issues/8))
```
Comment (Jean85) [inline CHANGELOG.md]:
```suggestion
* Force `typeMap['root']` to `BSONDocument` ([#8](https://github.com/facile-it/mongodb-messenger-transport/issues/8))
```

**Review by Jean85 — APPROVED** [review]
Just a small typo to fix, everything else is fine!

**Pull request merged**